source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -47,46 +46,62 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
 		return status;
 
 	MString materialXDocument;
-	MString elementName;
+	MString elementPath;
 	if( parser.isFlagSet(kDocumentFlag) )
 	{
 		argData.getFlagArgument(kDocumentFlag, 0, materialXDocument);
 	}
 	if (parser.isFlagSet(kElementFlag))
 	{
-		argData.getFlagArgument(kElementFlag, 0, elementName);
+		argData.getFlagArgument(kElementFlag, 0, elementPath);
 	}
-	if (materialXDocument.length() > 0 && elementName.length() > 0)
+	if (materialXDocument.length() > 0 && elementPath.length() > 0)
 	{
-		// Load document
-		MaterialX::DocumentPtr doc = MaterialX::createDocument();
-		MaterialX::readFromXmlFile(doc, materialXDocument.asChar());
+        std::unique_ptr<MaterialXData> materialXData{
+            new MaterialXData(materialXDocument.asChar(), elementPath.asChar())
+        };
 
-		// Load libraries
-		MaterialX::FilePath libSearchPath = Plugin::instance().getLibrarySearchPath();
-		const MaterialX::StringVec libraries = { "stdlib", "pbrlib", "bxdf", "stdlib/genglsl", "pbrlib/genglsl" };
-        MaterialX::loadLibraries(libraries, libSearchPath, doc);
-
-		// Check to make sure the elementName is a valid output node
-		if (!validOutputSpecified(doc, elementName.asChar()))
+		if (!materialXData->isValidOutput())
 		{
 			displayError("The element specified is not renderable.");
 			return MS::kFailure;
 		}
 
 		// Create the MaterialX node
-		MObject node = _dgModifier.createNode(MaterialXNode::MATERIALX_NODE_TYPEID);
+		MObject node = _dgModifier.createNode(
+            MaterialXTextureNode::MATERIALX_TEXTURE_NODE_TYPEID
+        );
 
 		// Generate a valid Maya node name from the path string
-        std::string nodeName = MaterialX::createValidName(elementName.asChar());
+		std::string nodeName = MaterialX::createValidName(elementPath.asChar());
 		_dgModifier.renameNode(node, nodeName.c_str());
 
-		std::string documentString = MaterialX::writeToXmlString(doc);
+		materialXData->createXMLWrapper();
+		materialXData->registerFragments();
+
+		MFnDependencyNode depNode(node);
+		auto materialXNode = dynamic_cast<MaterialXNode*>(depNode.userNode());
+        if (!materialXNode)
+        {
+            return MS::kFailure;
+        }
+
+        std::string documentString = MaterialX::writeToXmlString(materialXData->doc);
```

Comment:
For consistency I think we should use the namespace alias `namespace mx = MaterialX`. We do this elsewhere to avoid having to spell out the full MaterialX namespace everywhere, and it would be nice to follow that convention. (Pavlo this is not really part of your review as the full namespace was already used here before)